src/server/accounting.c

```diff
@@ -1577,237 +1261,121 @@ set_job_ProvAcctRcd(job *pjob, long time_se, int type)
  *			  PBS_ACCT_LAST.
  * @param[in]	buf	- buffer holding the data that will be stored in
  *			  accounting logs.
- * @param[in]	len	- number of characters in 'buf' still available to
- *			  store data.
- * @return	char *
- * @retval	pointer to 'buf' containing new data.
+ * @param[in]	len	- pointer to number of characters in 'buf' still 
+ * 			  available to store data.
+ * @return	void
  *
  */
-static char *
-build_common_data_for_job_update(job *pjob, int type, char *buf, int len)
+static void
+build_common_data_for_job_update(job *pjob, int type, char **buf, int *len)
 {
-	pbs_list_head attrlist;
-	int	  ct;
-	int       nd;
-	svrattrl *pal;
-	char 	*pb;
-	int	k, len_acct, att_index;
-	char	save_char;
-	int	old_perm;
+	pbs_list_head	attrlist;
+	svrattrl	*pal = NULL;
+	int		k = 0;
+	int		len_acct = 0;
+	int		att_index = 0;
+	char		save_char = 0;
+	int		old_perm = 0;
+	char		timestr[ACCTBUF_TIMES_NEED];
+#ifdef WIN32
+	char	*win_str = NULL;
+#endif
 
```

Comment:
Sort the variable declaration by it's type

Reply:
made the changes